internal/storage/bucket_handle.go

```diff
@@ -96,3 +98,38 @@ func (b *bucketHandle) StatObject(ctx context.Context, req *gcs.StatObjectReques
 
 	return
 }
+
+func (b *bucketHandle) CopyObject(ctx context.Context, req *gcs.CopyObjectRequest) (o *gcs.Object, err error) {
+	srcObj := b.bucket.Object(req.SrcName)
+	dstObj := b.bucket.Object(req.DstName)
+
+	// Switching to the requested Generation of Source Object.
+	if req.SrcGeneration != 0 {
+		srcObj = srcObj.Generation(req.SrcGeneration)
+	}
+
+	// Putting a condition that the MetaGeneration of source should match *req.SrcMetaGenerationPrecondition for copying operation to occur.
+	if req.SrcMetaGenerationPrecondition != nil {
+		srcObj = srcObj.If(storage.Conditions{MetagenerationMatch: *req.SrcMetaGenerationPrecondition})
+	}
+
+	objAttrs, err := dstObj.CopierFrom(srcObj).Run(ctx)
+
+	if err != nil {
+		switch ee := err.(type) {
+		case *googleapi.Error:
+			if ee.Code == http.StatusPreconditionFailed {
```

Comment:
did you verify if this casting is working?

Reply:
Yes it is working